core/object/undo_redo.cpp

```diff
@@ -254,9 +254,23 @@ void UndoRedo::commit_action() {
 		merging = false;
 	}
 
-	committing++;
-	redo(); // perform action
-	committing--;
+	if (p_execute) {
+		committing++;
+		redo(); // perform action
+		committing--;
+	} else {
+		// Similar to redo(), but without performing the action.
```

Comment:
Maybe you should make it `redo(p_execute)` to avoid the code duplication (or move the `redo` code to a new method and call it in `redo()`, if we don't want to mess with the bindings.

Otherwise there's the risk to introduce bugs if `redo()` is changed and not this code here.

Reply:
Yeah I did not wanted to expose the argument to the redo method, but I can indeed create a `_redo` method with the arg I'll change that.

Reply:
Done.